python/src/aac/io/files/find.py

```diff
@@ -0,0 +1,42 @@
+"""Module to comb directories and identify AaC files."""
+import logging
+import os
+
+from aac.io.parser import parse, ParserError
+from aac.io.files.aac_file import AaCFile
+
+
+def find_aac_files(root_directory_to_search: str) -> list[AaCFile]:
+    """Return a list of paths to AaC files in or under the root directory."""
+    aac_files = []
+
+    root_dir_abs_path = os.path.abspath(root_directory_to_search)
+    if os.path.isdir(root_dir_abs_path):
+        directory_results = os.walk(root_dir_abs_path)
+        for directory_path, directory_names, filenames in directory_results:
+            for filename in filenames:
+                filepath = os.path.join(directory_path, filename)
+                if is_aac_file(filepath):
+                    aac_files.append(AaCFile(filepath, True, False))
+    else:
+        logging.error(f"Root path '{root_dir_abs_path}' for AaC file discovery is not a directory.")
+
+    return aac_files
+
+
+def is_aac_file(filepath: str) -> bool:
+    """Test if a target file is considered a valid AaC file."""
+    filename, file_ext = os.path.splitext(filepath)
+
+    is_valid_aac_file = False
+    if file_ext == ".yaml" or file_ext == ".aac":
```

Comment:
They're defined as constants in the test files. I was on the edge about rearranging all of that since they were already a number of other changes in this PR